app/cdap/components/ConfigurationGroup/PropertyRow/index.tsx

```diff
@@ -82,7 +82,7 @@ const PLUGIN = 'plugin';
 const WIDGET_CATEGORY = 'widget-category';
 
 interface IState {
-  isMacroTextbox: boolean;
+  [key: string]: string | boolean;
```

Comment:
I had to make this change to keep TS happy about `[valueStateToBeUpdated]` on line 134. I can also keep the `Istate` interface as before and remove the redundant `setState` call like below:

```javascript
if (newValue) {
      this.setState({ isMacroTextbox: newValue, prevValue: this.props.value });
      this.handleChange(this.state.prevMacroValue);
    } else {
      this.setState({ isMacroTextbox: newValue, prevMacroValue: this.props.value });
      this.handleChange(this.state.prevValue);
    }
```